modules/core/src/effects/effect-preparator.js

```diff
@@ -0,0 +1,38 @@
+import LightingEffect from './lighting-effect';
+
+export default class EffectPreparator {
+  constructor(props) {}
+
+  prepareEffects({effects, layers}) {
+    const lightSources = this.prepareLightingEffects(effects);
+    return {
+      lightSources
+    };
+  }
+
+  prepareLightingEffects(effects) {
```

Comment:
It would be nice if the preparation was delegated to the actual lighting effect, instead of creating a class that knows about the internals of a specific effects.

```
class BaseEffect {
  prepareEffect() {} // do nothing
}

class LightingEffect extends Effect {
  prepareEffect() { ... } // do the stuff in this file
}

class EffectManager {
  prepareEffects() {
    for (effects in this.effects) {
      this.prepareEffect(...)
```

I do understand that not everything is easily generalized, sometimes it is a tradeoff.